### PR TITLE
Merkle tree parser

### PR DESCRIPTION
**Pull request by pablodeymo**

### CODE REVIEW

File: merkle_path/src/merkle_path.rs [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    fn to_bytes(self) -> Vec<u8> {
+        let mut ret = Vec::new();
+        for leaf in self {
+            match (leaf.left, leaf.right) {
+                (Some(left), None) => {
+                    let f = from_str(&left).unwrap();
+                    let bytes = to_bytes(&f);
+                    let padding_count = 32 - bytes.len();
+                    ret.extend(std::iter::repeat(0_u8).take(padding_count));
+                    for byte in bytes {
```
Comment (xqft) [inline merkle_path/src/merkle_path.rs:143]: This could be an `extend()`

File: merkle_path/src/merkle_path.rs [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+                    ret.extend(std::iter::repeat(0_u8).take(32));
+                }
+                (None, Some(right)) => {
+                    let f = from_str(&right).unwrap();
+                    let bytes = to_bytes(&f);
+                    let padding_count = 32 - bytes.len();
+                    ret.extend(std::iter::repeat(0_u8).take(padding_count));
+                    for byte in bytes {
+                        ret.push(byte);
+                    }
```
Comment (xqft) [inline merkle_path/src/merkle_path.rs:155]: This could be an `extend()`

**Review by xqft — APPROVED** [review]
despite the comments, lgtm

File: merkle_path/src/merkle_path.rs (hunk 1 above)
Comment (pablodeymo) [reply merkle_path/src/merkle_path.rs]: done

File: merkle_path/src/merkle_path.rs (hunk 2 above)
Comment (pablodeymo) [reply merkle_path/src/merkle_path.rs]: done

**Pull request merged**